internal/controller/postgrescluster/pgmonitor.go

```diff
@@ -456,7 +442,7 @@ func (r *Reconciler) reconcileExporterQueriesConfig(ctx context.Context,
 		return nil, err
 	}
 
-	if !pgmonitor.ExporterEnabled(cluster) {
+	if !pgmonitor.ExporterEnabled(ctx, cluster) || feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
```

Comment:
hmmm, `ExporterEnabled` returns false if OpenTelemetryMetrics is enabled, so we'd only check the part before the `||` here... Not sure if I have a question, just noticing.